### PR TITLE
VOOD-83: Add contributing information and support.

### PR DESCRIPTION
**Pull request by GinoCardillo-OST**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+extensions are `dbaeumer.vscode-eslint`, `esbenp.prettier-vscode` and `amodio.tsl-problem-matcher`. If you don't get
+the recommendations, install them manually from the Marketplace.
+
+Launch the extension by pressing F5 or by running the `Run Extension` launch configuration.
+
+## Testing
+
+### Adding Tests
+
+**TODO** General information about the test suite and how to format and structure tests.
```
Comment (GinoCardillo-OST) [inline CONTRIBUTING.md:60]: @alexandre-lagadec-ost Please add this section and check if I missed something in "Running Tests".

File: .prettierignore [hunk 2]
```diff
@@ -1,9 +1,12 @@
-# Based on .gitignore:
-out
-dist
-node_modules
-.vscode-test/
-*.vsix
-
-# Based on .eslintignore:
-#   N/A
+.github
```
Comment (alexandre-lagadec-ost) [inline .prettierignore:1]: Do we have to ignore .github completely?
Are the *.md files based on standard templates?
Does prettier corrupt the *.md files?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- Address a single concern in the least number of changed lines as possible.
+- Be accompanied by a complete Pull Request template (loaded automatically when a PR is created).
+
+In general, we follow the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr):
+
+1. Fork the repository to your own GitHub account
+2. Clone the project to your machine
+3. Create a branch locally with a succinct but descriptive name
+4. Commit changes to the branch
+5. Following the [formatting](#code-quality-tools) and [testing](#testing) guidelines
```
Comment (alexandre-lagadec-ost) [inline CONTRIBUTING.md:37]:
```suggestion
5. Following the [code quality](#code-quality-tools) and [testing](#testing) guidelines
```
Rationale: Code quality implies linting as well

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Testing
+
+### Adding Tests
+
+**TODO** General information about the test suite and how to format and structure tests.
+
+### Running Tests
+
+To run all test, execute the command `npm run test:ext`.
```
Comment (alexandre-lagadec-ost) [inline CONTRIBUTING.md:64]:
```suggestion
#### Installation and Integration

To run all extension installation and integration tests, execute the command `npm run test:ext`.
Use a virtual framebuffer such as Xvfb, if your environment does not provide a graphical interface (e. g. `xvfb-run -a npm run test:ext`).

#### Units

To run all unit tests, execute the command `npm run test:spec`.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexandre-lagadec-ost) [reply CONTRIBUTING.md:60]:
```suggestion
We use [mocha](https://mochajs.org/) as test runner library and [chai](https://www.chaijs.com/) as assertion library.

#### Installation and Integration

Installation and integration test files are expected to be located in the [test suite](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/tree/master/src/test/suite) directory and to have the file name extension `.test.ts` (e. g. [src/test/suite/extension.test.ts](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/blob/master/src/test/suite/extension.test.ts)).
Please refer to the [official VS Code extension testing documentation](https://code.visualstudio.com/api/working-with-extensions/testing-extension) for guidelines and examples.
Please use this type of tests to test migrations between versions and interactions with other extensions as well.

#### Units

Unit test files are expected to be located in the [source](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/tree/master/src) directory and to have the file name extension `.spec.ts` (e. g. [src/object-diagram/model/escapedString.spec.ts](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/blob/master/src/object-diagram/model/escapedString.spec.ts)).
In most cases, a unit test file tests an implementation located in the same directory with a file of the same name, but with the `.ts` file name extension (e. g. the file [src/object-diagram/model/escapedString.spec.ts](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/blob/master/src/object-diagram/model/escapedString.spec.ts) tests the implementations in file [src/object-diagram/model/escapedString.ts](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/blob/master/src/object-diagram/model/escapedString.ts)).
```

**Review by PascalSchuermannOst — APPROVED** [review]

File: .prettierignore (hunk 2 above)
Comment (GinoCardillo-OST) [reply .prettierignore:1]: Yes, yes and yes.

Comment (alexandre-lagadec-ost) [reply .prettierignore:1]: The workflow-related *.yml files are most probably not going to be corrupted by prettier. We could include the workflow-related *.yml files ... 

```suggestion
.github
!.github/workflows/
```

... which doesn't work due to [lacking negation capabilities][issue] or exclude just the template directories (preferred):

```suggestion
.github/*_TEMPLATE/
```

[issue]: https://github.com/prettier/prettier/issues/4912

**Pull request merged**